src/java.base/share/classes/java/lang/Object.java

```diff
@@ -152,6 +184,13 @@ public Object() {}
      * In other words, under the reference equality equivalence
      * relation, each equivalence class only has a single element.
      *
+     * <p>This method should <em>not</em> throw any exceptions during
+     * typical operation. In particular, {@link ClassCastException}
+     * should not be thrown if the argument has an incomparable type
+     * to this object and {@link NullPointerException} should not be
+     * thrown if the argument is {@code null}.  The implementation
```

Comment:
For these cases the recommendation should be to return false instead of throwing such exceptions.

Reply:
Good point; updated accordingly.